### PR TITLE
feat(react): update icons to next version

### PR DESCRIPTION
**Pull request by joshblack**
This PR updates our codemod, migration docs, and our codebase to the new icon API.

#### Changelog

**New**

**Changed**

- Update React codebase with new icons
- Update our codemod to handle some situations that came up when running against our codebase
- Update our migration docs with more manual steps / instructions

**Removed**

#### Testing / Reviewing

- Verify that icons in Storybook render/look as expected (typically the 16x16 variant of the icon)
- Verify stories in Storybook that use icons work as expected (like in Button)

### CODE REVIEW

File: packages/react/src/components/Button/Button-story.js [hunk 1]
```diff
@@ -35,11 +34,12 @@ const icons = {
...
   iconSearch,
-  Add16,
-  AddFilled16,
-  Search16,
-  PlayOutlineFilled16,
-  PlayOutlineFilled32,
+  Add16: Add,
+  AddFilled16: AddFilled,
+  Search16: Search,
+  PlayOutlineFilled16: PlayOutlineFilled,
```
Comment (emyarod) [inline packages/react/src/components/Button/Button-story.js:40]: preview:

image

`main`:

image

File: packages/react/src/components/TreeView/TreeView-story.js [hunk 2]
```diff
@@ -6,7 +6,7 @@
  */
 
 import React, { useState } from 'react';
-import { Document16, Folder16 } from '@carbon/icons-react';
+import { Document, Folder } from '@carbon/icons-react/next';
```
Comment (emyarod) [inline packages/react/src/components/TreeView/TreeView-story.js:9]: preview:

image

`main`:

image

File: packages/react/src/components/UIShell/HeaderMenuButton.js [hunk 3]
```diff
@@ -38,8 +38,8 @@ const HeaderMenuButton = ({
     'aria-label': ariaLabel,
     'aria-labelledby': ariaLabelledBy,
   };
-  const menuIcon = renderMenuIcon ? renderMenuIcon : <Menu20 />;
-  const closeIcon = renderCloseIcon ? renderCloseIcon : <Close20 />;
+  const menuIcon = renderMenuIcon ? renderMenuIcon : <Menu size={20} />;
+  const closeIcon = renderCloseIcon ? renderCloseIcon : <Close size={20} />;
```
Comment (emyarod) [inline packages/react/src/components/UIShell/HeaderMenuButton.js:42]:
```suggestion
  const menuIcon = renderMenuIcon || <Menu size={20} />;
  const closeIcon = renderCloseIcon || <Close size={20} />;
```

a simpler alternative maybe

File: packages/react/src/components/Button/Button-story.js [hunk 4]
```diff
@@ -35,11 +34,12 @@ const icons = {
...
   iconAddSolid,
   iconSearch,
-  Add16,
-  AddFilled16,
-  Search16,
-  PlayOutlineFilled16,
-  PlayOutlineFilled32,
+  Add16: Add,
+  AddFilled16: AddFilled,
+  Search16: Search,
```
Comment (emyarod) [inline packages/react/src/components/Button/Button-story.js:39]: preview:

image

`main`:

image

File: packages/react/src/components/DatePickerInput/DatePickerInput.js [hunk 5]
```diff
@@ -10,10 +10,10 @@ import React, { Component } from 'react';
 import classNames from 'classnames';
 import { settings } from 'carbon-components';
 import {
-  Calendar16,
-  WarningFilled16,
-  WarningAltFilled16,
-} from '@carbon/icons-react';
+  Calendar,
+  WarningFilled,
```
Comment (emyarod) [inline packages/react/src/components/DatePickerInput/DatePickerInput.js:14]: preview:

image

`main`:

image

File: packages/react/src/components/TextInput/PasswordInput.js [hunk 6]
```diff
@@ -3,11 +3,11 @@ import classNames from 'classnames';
 import PropTypes from 'prop-types';
 import { settings } from 'carbon-components';
 import {
-  View16,
-  ViewOff16,
-  WarningAltFilled16,
-  WarningFilled16,
-} from '@carbon/icons-react';
+  View,
+  ViewOff,
```
Comment (emyarod) [inline packages/react/src/components/TextInput/PasswordInput.js:7]: preview:

image

image

`main`:

image

image

**Review by emyarod — COMMENTED** [review]
should we add a visual review to this PR as well?

I only left a comment on one instance of each broken icon but they appear to be broken across the board except for `WarningFilled` which had some variation

**Comment by joshblack** [comment]
Going to close this one out since it'll be a pain to deal with merge conflicts 😂  Going to take a look at @emyarod's notes above to see what's going on and re-open 👀

**Pull request closed without merging**